cmd/rollkit/commands/run_node.go

```diff
@@ -213,40 +214,91 @@ func NewRunNodeCmd() *cobra.Command {
 				return fmt.Errorf("failed to create new rollkit node: %w", err)
 			}
 
-			// Start the node
-			if err := rollnode.Start(ctx); err != nil {
+			// Create error channel and signal channel
+			errCh := make(chan error, 1)
+			shutdownCh := make(chan struct{})
+
+			// Start the node in a goroutine
+			go func() {
+				err := rollnode.Run(ctx)
+				select {
+				case errCh <- err:
+				default:
+					logger.Error("Error channel full", "error", err)
+				}
```

Comment:
probably could add panic recovery as well as an option

Reply:
do we panic as a form of error in the code?

Reply:
not during runtime